diesel_tests/tests/filter.rs

```diff
@@ -1,6 +1,14 @@
 use schema::*;
 use diesel::*;
 
+macro_rules! assert_sets_eq {
+    ($set1:expr, $set2:expr) => {
+        let set1 = {$set1};
+        let set2 = {$set2};
+        assert!(set1.iter().all(|si| set2.contains(si)) && set2.iter().all(|si| set1.contains(si)));
```

Comment:
Might be easier to convert them to HashSets and call `.difference().len() == 0`.